Build failure compiling Godot for Windows with LLVM MinGW (missing file)

### Tested versions

- Reproducible in master [8e2141e]
- Not reproducible in Godot 4.3.dev6 [64520fe]

### System information

Godot v4.3.beta (2311e0462) - Windows 10.0.22631 - Vulkan (Forward+) - dedicated Radeon RX 580 Series (Advanced Micro Devices, Inc.; 31.0.21912.14) - Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz (6 Threads)

### Issue description

When I try to compile Godot for windows using LLVM MinGW, before reaching the `Generating ...` stages, it silently fails the build.
```bash
Building editor for x86_64. Build may take longer.
scons: Reading SConscript files ...
Using MinGW, arch x86_64
Building for platform "windows", architecture "x86_64", target "editor".
Checking for C header file mntent.h... no
scons: done reading SConscript files.
scons: Building targets ...
Copy("bin/D3D12Core.dll", "/home/mbcx/Documents/d3d12/agilitysdk/build/native/bin/x64/D3D12Core.dll")
Copy("bin/WinPixEventRuntime.dll", "/home/mbcx/Documents/d3d12/pix/bin/x64/WinPixEventRuntime.dll")
Copy("bin/d3d12SDKLayers.dll", "/home/mbcx/Documents/d3d12/agilitysdk/build/native/bin/x64/d3d12SDKLayers.dll")
Copy("bin/dxil.dll", "/home/mbcx/Documents/d3d12/dxc/bin/x64/dxil.dll")
BUILD: Elapsed time [00:00:04]
Compiling platform/windows/console_wrapper_windows.cpp ...
build_res_file(["platform/windows/godot_res_wrap.windows.editor.x86_64.llvm.o"], ["platform/windows/godot_res_wrap.rc"])
Generating core/disabled_classes.gen.h ...
scons: *** [platform/windows/godot_res_wrap.windows.editor.x86_64.llvm.o] Error -1
```
I found out that [this commit is the culprit](https://github.com/godotengine/godot/pull/91624/commits/34fb3f7730e93120c0f52cc86f314d50fa9d34ed). When I run the command manually, this is causing the build to fail.
```bash
/home/mbcx/sysroot/llvm-mingw/bin/x86_64-w64-mingw32-windres --include-dir . --target=pe-x86-64 -i platform/windows/godot_res_wrap.rc -o platform/windows/godot_res_wrap.windows.editor.x86_64.llvm.o
In file included from platform/windows/godot_res_wrap.rc:1:
./core/version.h:34:10: fatal error: 'core/version_generated.gen.h' file not found
   34 | #include "core/version_generated.gen.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
llvm-rc: Preprocessing failed.
```

### Steps to reproduce

I'm not sure if this is a LLVM MinGW specific bug.
1. Download the latest [LLVM MinGW](https://github.com/mstorsjo/llvm-mingw/releases)
2. Specify it in your `MINGW_PREFIX` environment variable.
3. As I build Godot through a script, this is the full command.
```bash
scons -j${THREADS_PER_JOB} platform="windows" precision="single" \ # $os $precision
    use_mingw=yes use_llvm=yes linker=mold lto=none module_text_server_fb_enabled=yes \
    arch=x86_64 d3d12=yes agility_sdk_multi_arch=yes use_pix=yes \
    dxc_path=$HOME/Documents/d3d12/dxc \
    mesa_libs=$HOME/Documents/nir/win \
    agility_sdk_path=$HOME/Documents/d3d12/agilitysdk \
    pix_path=$HOME/Documents/d3d12/pix \
    angle_libs=$HOME/sources/godot-angle-static/bin
```
4. Wait for failure

### Minimal reproduction project (MRP)

N/A

--- a/platform/windows/SCsub
+++ b/platform/windows/SCsub
@@ -50,6 +50,7 @@ def arrange_program_clean(prog):
 res_file = "godot_res.rc"
 res_target = "godot_res" + env["OBJSUFFIX"]
 res_obj = env.RES(res_target, res_file)
+env.Depends(res_obj, "#core/version_generated.gen.h")
 
 env.add_source_files(sources, common_win)
 sources += res_obj
@@ -63,6 +64,7 @@ if env["windows_subsystem"] == "gui":
     res_wrap_file = "godot_res_wrap.rc"
     res_wrap_target = "godot_res_wrap" + env["OBJSUFFIX"]
     res_wrap_obj = env_wrap.RES(res_wrap_target, res_wrap_file)
+    env_wrap.Depends(res_wrap_obj, "#core/version_generated.gen.h")
 
     if env.msvc:
         env_wrap.Append(LINKFLAGS=["/SUBSYSTEM:CONSOLE"])

